### DISCUSSION TITLE
SD CARD Mount Sample Code

### DISCUSSION THREAD

**Discussion by spienz** [lvgl-micropython/lvgl_micropython#162] [General]
Hey Folks,
as diskussed with @kdschlosser some sample for mounting SDCARD.

Greetings

Sebastian

```python
from machine import SPI, SDCard
import vfs


# Initialize the SPI bus with appropriate pins
spi_bus = SPI.Bus(
    host=2,        # SPI bus host number (adjust based on your board)
    mosi=11,       # Master Out Slave In (adjust based on your hardware)
    miso=13,       # Master In Slave Out (adjust based on your hardware)
    sck=12         # Clock pin (adjust based on your hardware)
)


sd = None

while sd is None:
    try:
        sd = machine.SDCard(
            spi_bus=spi_bus,
            cs=10,
            freq=1000000
        )
        vfs.mount(sd, "/sd")
        print('Flash Mounted')
        print(os.listdir('/sd'))
    except Exception as er:
        sd = None
        print('sd card failure')
        time.sleep(5)

```

**Comment by anhthypp** [comment]
I upload firmware to ESP32 LCD from
python3 make.py esp32 --partition-size=2560000 BOARD=ESP32_GENERIC DISPLAY=ILI9341 INDEV=XPT2046

And I tested this sample code on ESP32 LCD but it restart with error
from machine import SPI, SDCard
import vfs
import time
import os

# Initialize the SPI bus with appropriate pins
spi_bus = SPI.Bus(
    host=2,        # SPI bus host number (adjust based on your board)
    mosi=23,       # Master Out Slave In (adjust based on your hardware)
    miso=19,       # Master In Slave Out (adjust based on your hardware)
    sck=18         # Clock pin (adjust based on your hardware)
)

sd = None

sd = SDCard(
            spi_bus=spi_bus,
            cs=5,
            freq=1000000
        )

vfs.mount(sd, "/sd")
print('Flash Mounted')
print(os.listdir('/sd'))

--------
LVGL (9.2.2) MicroPython (1.25.0) Binding compiled on 2025-06-20; Generic ESP32 module with ESP32

Type "help()" for more information.

>>> %Run -c $EDITOR_CONTENT

MPY: soft reboot
Traceback (most recent call last):
  File "<stdin>", line 22, in <module>
OSError: 16
>>>
